<a href="https://colab.research.google.com/github/tmarissa/Adv_DataAnalytics/blob/master/ipynb/501_Analysis_FIPS.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# DATA 606 Capstone
## Marissa Tan
### Impact of COVID-19 on the US Housing Market
__Real Estate and Density Dataset__

This notebook performs analysis on 2019 and 2021 for:
- FIPS - whole United States
    - Average List Price Mean
    - Average List Price Change Mean
- Individual States - Arizona and Massachusetts
    - Average List Price Mean
    - Average List Price Change Mean
        
Reference:<br>
Maede Maftouni of Kaggle. Link: https://www.kaggle.com/maedemaftouni/us-real-estate-market-trends-visualization<br>

https://datascientyst.com/get-top-10-highest-lowest-values-pandas/#:~:text=%20How%20to%20Get%20Top%2010%20Highest%20or,you%20can%20see%20in%20Step%202...%20More%20

In [1]:
import pandas as pd
import numpy as np
import plotly.express as px
import matplotlib
import matplotlib.pyplot as plt
import seaborn as sns

# 1. 2019 and 2021 CSV
This notebook will evaluate and compare the average listing price and average listing price change on the state and county level. 

## 1.1 Read 2019 and 2021 cleansed csv

In [2]:
df_2019 = pd.read_csv('df_2019.csv', index_col=False)
df_2019.head(3)

FIPS  year state         county  density  average_listing_price  \
0  4001  2019    AZ  Apache County        5               427761.0   
1  4001  2019    AZ  Apache County        5               436694.0   
2  4001  2019    AZ  Apache County        5               451358.0   

   average_listing_price_mm  average_listing_price_yy  median_listing_price  \
0                    0.1306                    0.1849              243500.0   
1                    0.0209                    0.2343              243500.0   
2                    0.0336                    0.3307              246714.0   

   median_listing_price_mm  median_listing_price_yy    rural_%  rural_cat  \
0                   0.0188                   0.1195  74.061076          1   
1                   0.0000                   0.1068  74.061076          1   
2                   0.0132                   0.1637  74.061076          1   

   total_listing_count  
0                178.0  
1                183.0  
2                182.0

In [3]:
df_2021 = pd.read_csv('df_2021.csv', index_col=False)
df_2021.head(3)

FIPS  year state         county  density  average_listing_price  \
0  4001  2021    AZ  Apache County        5               495914.0   
1  4001  2021    AZ  Apache County        5               491306.0   
2  4001  2021    AZ  Apache County        5               561300.0   

   average_listing_price_mm  average_listing_price_yy  median_listing_price  \
0                   -0.0197                    0.1282              279000.0   
1                   -0.0093                    0.1099              276750.0   
2                    0.1425                    0.0189              279000.0   

   median_listing_price_mm  median_listing_price_yy    rural_%  rural_cat  \
0                  -0.0016                  -0.0571  74.061076          1   
1                  -0.0081                  -0.0650  74.061076          1   
2                   0.0081                  -0.1143  74.061076          1   

   total_listing_count  
0                 79.0  
1                 76.0  
2                 76.0

## 2. FIPS


## 2.1 Average Listing Price (Mean)

### 2.1a For the Year 2019

In [4]:
# This 10 most expensive listing price by FIPS (state and county code) in 2019
df_2019_FIPS = df_2019.groupby(['FIPS', 'state', 'county', 'density', 
                               'rural_%', 'rural_cat'])['average_listing_price'].mean().nlargest(10).reset_index()
df_2019_FIPS.head(5)

FIPS state           county  density    rural_%  rural_cat  \
0  15007    HI     Kauai County      118  12.860145          2   
1  15009    HI      Maui County      141  14.567860          2   
2  25025    MA   Suffolk County    13722   0.074091          2   
3  15003    HI  Honolulu County     1692   0.862877          2   
4  44005    RI   Newport County      836  12.092221          2   

   average_listing_price  
0           2.022722e+06  
1           2.000211e+06  
2           1.471328e+06  
3           1.154390e+06  
4           1.097214e+06

### 3.1b For the Year 2021

In [5]:
# The 10 most expensive listing price according to FIPS (county and state code) in 2021  
df_2021_FIPS = df_2021.groupby(['FIPS', 'state', 'county', 'density', 'rural_%', 
                                'rural_cat'])['average_listing_price'].mean().nlargest(10).reset_index()
df_2021_FIPS.head(5)

FIPS state             county  density    rural_%  rural_cat  \
0  15007    HI       Kauai County      118  12.860145          2   
1  15009    HI        Maui County      141  14.567860          2   
2  44005    RI     Newport County      836  12.092221          2   
3  25025    MA     Suffolk County    13722   0.074091          2   
4  25001    MA  Barnstable County      581   7.483047          2   

   average_listing_price  
0           2.273151e+06  
1           2.261603e+06  
2           1.641150e+06  
3           1.582755e+06  
4           1.476872e+06

### 2.1c Similarities Average List Price (2019 and 2021)

In [6]:
# Similar FIPS with a high increase from the top 10 list of 2019 and 2021
similar = (df_2021_FIPS['FIPS']).isin(df_2019_FIPS['FIPS'])
df_2021_FIPS[similar]

FIPS state             county  density    rural_%  rural_cat  \
0  15007    HI       Kauai County      118  12.860145          2   
1  15009    HI        Maui County      141  14.567860          2   
2  44005    RI     Newport County      836  12.092221          2   
3  25025    MA     Suffolk County    13722   0.074091          2   
4  25001    MA  Barnstable County      581   7.483047          2   
5  15003    HI    Honolulu County     1692   0.862877          2   
6  15001    HI      Hawaii County       49  37.990804          2   

   average_listing_price  
0           2.273151e+06  
1           2.261603e+06  
2           1.641150e+06  
3           1.582755e+06  
4           1.476872e+06  
5           1.288058e+06  
6           1.283354e+06

Differences 2019 and 2021

In [7]:
df_2019_FIPS[~(df_2019_FIPS['FIPS']).isin(df_2021_FIPS['FIPS'])]

FIPS state                county  density   rural_%  rural_cat  \
6  25017    MA      Middlesex County     1995  3.027706          2   
7  11001    DC  District of Columbia    11294  0.000000          2   
9  25021    MA        Norfolk County     1832  1.240489          2   

   average_listing_price  
6          977263.083333  
7          970809.750000  
9          961562.333333

In [8]:
df_2021_FIPS[~(df_2021_FIPS['FIPS']).isin(df_2019_FIPS['FIPS'])]

FIPS state           county  density    rural_%  rural_cat  \
7  8067    CO  La Plata County       32  59.948572          1   
8  8013    CO   Boulder County      455   8.913083          2   
9  8051    CO  Gunnison County        5  58.607413          1   

   average_listing_price  
7              1247796.0  
8              1240170.0  
9              1231790.0

The consistently high average listing price based on FIPS in 2019 and 2021 are Kauai County and Maui County found in Hawaii. In 2019, it is followed by Suffolk County (Massahussets), Honolulu County (HI), and Newport (RI). However, in 2021, Newport (RI) in 5th moved to 3rd place which displaced a a higher density and lower rural percentage Suffolk (MA). In 2021, Barnstable (MA) is in 5th place displaced Honolulu (HI) which has a higher density and lower rural percentage than Barnstable (MA).


## 2.2 Change Average Listing Price (Mean)
This will group the FIPS uniquely and get the mean of the average listing price


### 2.2a For the Year 2019

In [9]:
# The 10 biggest change average listing price according to FIPS (county and state code) in 2019
FIPS_yy_2019 = df_2019.groupby(['FIPS', 'state', 'county', 'density', 
                                'rural_%', 'rural_cat'])['average_listing_price_yy'].mean().nlargest(5).reset_index()
FIPS_yy_2019

FIPS state               county  density  rural_%  rural_cat  \
0  40057    OK        Harmon County        4    100.0          1   
1  31129    NE      Nuckolls County        7    100.0          1   
2  48433    TX     Stonewall County        1    100.0          1   
3  30075    MT  Powder River County        0    100.0          1   
4  13125    GA      Glascock County       20    100.0          1   

   average_listing_price_yy  
0                  8.233333  
1                  8.007422  
2                  6.217400  
3                  5.036933  
4                  4.942773

### 2.2b For the Year 2021

In [10]:
# The 10 biggest change average listing price according to FIPS (county and state code) in 2021
FIPS_yy_2021 = df_2021.groupby(['FIPS', 'state', 'county', 'density', 
                                'rural_%', 'rural_cat'])['average_listing_price_yy'].mean().nlargest(5).reset_index()
FIPS_yy_2021

FIPS state               county  density     rural_%  rural_cat  \
0  16033    ID         Clark County        0  100.000000          1   
1  48433    TX     Stonewall County        1  100.000000          1   
2  38087    ND         Slope County        0  100.000000          1   
3   4011    AZ      Greenlee County        5   46.568686          2   
4   2060    AK  Bristol Bay Borough        1  100.000000          1   

   average_listing_price_yy  
0                  5.430067  
1                  4.896329  
2                  4.180000  
3                  3.247392  
4                  3.040167

### 2.2c Similarities Average List Price Mean (2019 and 2021) 
This uses top 50 results from the 2019 and 2021 

In [11]:
# Similar FIPS with a high mean change in average listing price from the top 10 list of 2019 and 2021
similar =(FIPS_yy_2021['FIPS']).isin(FIPS_yy_2019['FIPS'])
FIPS_yy_2021[similar]

FIPS state            county  density  rural_%  rural_cat  \
1  48433    TX  Stonewall County        1    100.0          1   

   average_listing_price_yy  
1                  4.896329

## 2.3 Change Average Listing Price 


### 2.3a For the Year 2019

In [12]:
df_2019.describe()

FIPS     year       density  average_listing_price  \
count  30507.000000  30507.0  30507.000000           3.050700e+04   
mean   30460.022585   2019.0    185.483364           2.509845e+05   
std    15115.862425      0.0    952.428290           1.355666e+05   
min     1001.000000   2019.0      0.000000           4.632000e+04   
25%    19003.000000   2019.0     16.000000           1.654610e+05   
50%    29167.000000   2019.0     40.000000           2.202060e+05   
75%    45079.000000   2019.0     99.000000           2.990315e+05   
max    56045.000000   2019.0  34979.000000           2.235543e+06   

       average_listing_price_mm  average_listing_price_yy  \
count              30475.000000              30402.000000   
mean                   0.007495                  0.076221   
std                    0.127627                  0.494169   
min                   -0.939100                 -0.990300   
25%                   -0.021200                 -0.039500   
50%                    0.002200                  0.039900   
75%                    0.028600                  0.132300   
max                   12.300000                 39.000000   

       median_listing_price  median_listing_price_mm  median_listing_price_yy  \
count          3.050700e+04             30475.000000             30402.000000   
mean           1.959187e+05                 0.010767                 0.087112   
std            9.545046e+04                 0.198717                 0.511591   
min            2.150000e+04                -0.905900                -0.975100   
25%            1.304290e+05                -0.022400                -0.033600   
50%            1.734000e+05                 0.000000                 0.045900   
75%            2.399000e+05                 0.029950                 0.138500   
max            1.149000e+06                24.600000                39.000000   

            rural_%     rural_cat  total_listing_count  
count  30507.000000  30507.000000         30507.000000  
mean      60.047718      1.379126           393.214377  
std       30.177180      0.485178           992.919636  
min        0.000000      1.000000             0.000000  
25%       36.822160      1.000000            44.000000  
50%       60.720947      1.000000           122.000000  
75%       86.744724      2.000000           328.000000  
max      100.000000      2.000000         26147.000000

In [13]:
df_2019

FIPS  year state           county  density  average_listing_price  \
0       4001  2019    AZ    Apache County        5               427761.0   
1       4001  2019    AZ    Apache County        5               436694.0   
2       4001  2019    AZ    Apache County        5               451358.0   
3       4001  2019    AZ    Apache County        5               469451.0   
4       4001  2019    AZ    Apache County        5               459048.0   
...      ...   ...   ...              ...      ...                    ...   
30502  33019  2019    NH  Sullivan County       80               365495.0   
30503  33019  2019    NH  Sullivan County       80               383805.0   
30504  33019  2019    NH  Sullivan County       80               365505.0   
30505  33019  2019    NH  Sullivan County       80               337347.0   
30506  33019  2019    NH  Sullivan County       80               316672.0   

       average_listing_price_mm  average_listing_price_yy  \
0                        0.1306                    0.1849   
1                        0.0209                    0.2343   
2                        0.0336                    0.3307   
3                        0.0401                    0.3989   
4                       -0.0222                    0.3801   
...                         ...                       ...   
30502                    0.0569                    0.0794   
30503                    0.0501                    0.1957   
30504                   -0.0652                    0.0023   
30505                   -0.0770                   -0.0160   
30506                   -0.0613                    0.0231   

       median_listing_price  median_listing_price_mm  median_listing_price_yy  \
0                  243500.0                   0.0188                   0.1195   
1                  243500.0                   0.0000                   0.1068   
2                  246714.0                   0.0132                   0.1637   
3                  262714.0                   0.0649                   0.1942   
4                  262000.0                  -0.0027                   0.1936   
...                     ...                      ...                      ...   
30502              254900.0                   0.0439                   0.1586   
30503              269000.0                   0.0553                   0.1850   
30504              249950.0                  -0.0179                  -0.0177   
30505              246000.0                  -0.0158                   0.0254   
30506              244200.0                  -0.0073                   0.0308   

         rural_%  rural_cat  total_listing_count  
0      74.061076          1                178.0  
1      74.061076          1                183.0  
2      74.061076          1                182.0  
3      74.061076          1                192.0  
4      74.061076          1                217.0  
...          ...        ...                  ...  
30502  64.162590          1                282.0  
30503  64.162590          1                322.0  
30504  64.162590          1                373.0  
30505  64.162590          1                327.0  
30506  64.162590          1                262.0  

[30507 rows x 14 columns]

In [14]:
# This is the 10 biggest change in the average lisitng price in a county and state (FIPS)
## at any point in time in 2019
df_2019.drop_duplicates('average_listing_price_yy').nlargest(10, 'average_listing_price_yy')

FIPS  year state               county  density  average_listing_price  \
951     2185  2019    AK  North Slope Borough        0               300000.0   
4318   13125  2019    GA      Glascock County       20               260750.0   
17097  31129  2019    NE      Nuckolls County        7               180650.0   
9402   20157  2019    KS      Republic County        6               138470.0   
17096  31129  2019    NE      Nuckolls County        7               177102.0   
17093  31129  2019    NE      Nuckolls County        7               265733.0   
19001  40057  2019    OK        Harmon County        4               140000.0   
17094  31129  2019    NE      Nuckolls County        7               173144.0   
20949  46003  2019    SD        Aurora County        3               249183.0   
19003  40057  2019    OK        Harmon County        4               133000.0   

       average_listing_price_mm  average_listing_price_yy  \
951                     -0.0909                   39.0000   
4318                     0.5340                   19.8600   
17097                    0.0200                   19.2978   
9402                     0.0336                   19.0682   
17096                    0.2618                   18.8991   
17093                   -0.4684                   14.2611   
19001                    0.3333                   13.0000   
17094                   -0.3484                   12.6334   
20949                   -0.0057                   12.4693   
19003                    0.0472                   12.3000   

       median_listing_price  median_listing_price_mm  median_listing_price_yy  \
951                300000.0                  -0.0909                  39.0000   
4318               339000.0                   0.5860                  26.1200   
17097              120000.0                   0.5000                  12.4831   
9402                84800.0                   1.8361                  11.2899   
17096               80000.0                  -0.0244                   7.9888   
17093              322400.0                  -0.3551                  17.5154   
19001              140000.0                   0.3333                  13.0000   
17094              117043.0                  -0.6370                   8.2160   
20949              149900.0                   0.0000                   7.1027   
19003              133000.0                  -0.0292                  12.3000   

          rural_%  rural_cat  total_listing_count  
951     59.331919          1                  0.0  
4318   100.000000          1                  6.0  
17097  100.000000          1                 23.0  
9402   100.000000          1                  5.0  
17096  100.000000          1                 18.0  
17093  100.000000          1                  2.0  
19001  100.000000          1                  1.0  
17094  100.000000          1                  8.0  
20949  100.000000          1                  4.0  
19003  100.000000          1                  1.0

In [15]:
df_2019.describe()

FIPS     year       density  average_listing_price  \
count  30507.000000  30507.0  30507.000000           3.050700e+04   
mean   30460.022585   2019.0    185.483364           2.509845e+05   
std    15115.862425      0.0    952.428290           1.355666e+05   
min     1001.000000   2019.0      0.000000           4.632000e+04   
25%    19003.000000   2019.0     16.000000           1.654610e+05   
50%    29167.000000   2019.0     40.000000           2.202060e+05   
75%    45079.000000   2019.0     99.000000           2.990315e+05   
max    56045.000000   2019.0  34979.000000           2.235543e+06   

       average_listing_price_mm  average_listing_price_yy  \
count              30475.000000              30402.000000   
mean                   0.007495                  0.076221   
std                    0.127627                  0.494169   
min                   -0.939100                 -0.990300   
25%                   -0.021200                 -0.039500   
50%                    0.002200                  0.039900   
75%                    0.028600                  0.132300   
max                   12.300000                 39.000000   

       median_listing_price  median_listing_price_mm  median_listing_price_yy  \
count          3.050700e+04             30475.000000             30402.000000   
mean           1.959187e+05                 0.010767                 0.087112   
std            9.545046e+04                 0.198717                 0.511591   
min            2.150000e+04                -0.905900                -0.975100   
25%            1.304290e+05                -0.022400                -0.033600   
50%            1.734000e+05                 0.000000                 0.045900   
75%            2.399000e+05                 0.029950                 0.138500   
max            1.149000e+06                24.600000                39.000000   

            rural_%     rural_cat  total_listing_count  
count  30507.000000  30507.000000         30507.000000  
mean      60.047718      1.379126           393.214377  
std       30.177180      0.485178           992.919636  
min        0.000000      1.000000             0.000000  
25%       36.822160      1.000000            44.000000  
50%       60.720947      1.000000           122.000000  
75%       86.744724      2.000000           328.000000  
max      100.000000      2.000000         26147.000000

### 2.3b For the Year 2021

In [16]:
df_2021.describe()

FIPS     year       density  average_listing_price  \
count  30528.000000  30528.0  30528.000000           3.052800e+04   
mean   30351.350301   2021.0    180.900976           3.075573e+05   
std    15101.151588      0.0    667.790422           1.848283e+05   
min     1001.000000   2021.0      0.000000           1.590000e+04   
25%    19007.000000   2021.0     16.000000           1.884002e+05   
50%    29137.000000   2021.0     41.000000           2.683070e+05   
75%    45053.000000   2021.0    102.000000           3.743728e+05   
max    56045.000000   2021.0  15692.000000           2.850052e+06   

       average_listing_price_mm  average_listing_price_yy  \
count              30453.000000              30384.000000   
mean                   0.017169                  0.166224   
std                    0.179640                  0.400101   
min                   -0.989600                 -0.986900   
25%                   -0.039700                 -0.015200   
50%                    0.005000                  0.117600   
75%                    0.055000                  0.279600   
max                   10.653100                 10.920400   

       median_listing_price  median_listing_price_mm  median_listing_price_yy  \
count          3.052800e+04             30453.000000             30384.000000   
mean           2.338997e+05                 0.036315                 0.163386   
std            1.257422e+05                 1.811652                 0.453935   
min            2.382000e+03                -0.994700                -0.994700   
25%            1.489000e+05                -0.039700                -0.029900   
50%            2.049000e+05                 0.000000                 0.096900   
75%            2.890000e+05                 0.056300                 0.254200   
max            1.625000e+06               313.441600                12.428600   

            rural_%     rural_cat  total_listing_count  
count  30528.000000  30528.000000         30528.000000  
mean      59.618691      1.384041           274.228806  
std       30.469645      0.486376           714.941248  
min        0.000000      1.000000             0.000000  
25%       35.527833      1.000000            26.000000  
50%       60.357170      1.000000            74.000000  
75%       86.849780      2.000000           210.000000  
max      100.000000      2.000000         15861.000000

In [17]:
# This is the 10 biggest change in the average lisitng price in a county and state (FIPS)
# at any point in time in 2021
df_2021.drop_duplicates('average_listing_price_yy').nlargest(10, 'average_listing_price_yy')

FIPS  year state            county  density  average_listing_price  \
71      4011  2021    AZ   Greenlee County        5               893580.0   
23714  48433  2021    TX  Stonewall County        1               207000.0   
30333  16033  2021    ID      Clark County        0               512483.0   
23713  48433  2021    TX  Stonewall County        1               207000.0   
23712  48433  2021    TX  Stonewall County        1               197500.0   
22086  48107  2021    TX     Crosby County        5               266833.0   
23711  48433  2021    TX  Stonewall County        1               194975.0   
30332  16033  2021    ID      Clark County        0               449967.0   
2555    8017  2021    CO   Cheyenne County        0               224933.0   
70      4011  2021    AZ   Greenlee County        5              1151407.0   

       average_listing_price_mm  average_listing_price_yy  \
71                      -0.2239                   10.9204   
23714                    0.0000                    8.2000   
30333                    0.1389                    7.5414   
23713                    0.0481                    7.2800   
23712                    0.0130                    6.9000   
22086                    1.8311                    6.8654   
23711                    0.0000                    6.7990   
30332                    0.0000                    6.4994   
2555                     0.1252                    6.4978   
70                       0.2636                    6.4085   

       median_listing_price  median_listing_price_mm  median_listing_price_yy  \
71                 160000.0                  -0.3600                   1.1333   
23714              279000.0                   0.0000                  11.4000   
30333              575000.0                   0.6429                   8.5833   
23713              279000.0                   0.3007                  10.1600   
23712              214500.0                  -0.1061                   7.5800   
22086              307000.0                   1.7301                   4.7979   
23711              239950.0                   0.0000                   8.5980   
30332              350000.0                   0.0000                   4.8333   
2555               175000.0                   0.7518                   4.8333   
70                 250000.0                   0.0000                   1.7174   

          rural_%  rural_cat  total_listing_count  
71      46.568686          2                  8.0  
23714  100.000000          1                  2.0  
30333  100.000000          1                  6.0  
23713  100.000000          1                  3.0  
23712  100.000000          1                  3.0  
22086  100.000000          1                  7.0  
23711  100.000000          1                  3.0  
30332  100.000000          1                  6.0  
2555   100.000000          1                  4.0  
70      46.568686          2                  6.0

Without using the mean for the average listing price change, it is unable to see a pattern because to the duplication in FIPS. Best to disregard this analysis

# 3. Individual States (Arizona)


## 3.1 Average Listing Price

### 3.1a For the Year 2019

In [18]:
df_2019[(df_2019['state']=='AZ')].describe().T

count           mean            std          min  \
FIPS                      180.0    4013.866667       7.826613    4001.0000   
year                      180.0    2019.000000       0.000000    2019.0000   
density                   180.0      57.333333     117.154341       3.0000   
average_listing_price     180.0  409346.800000  144641.223262  184000.0000   
average_listing_price_mm  180.0       0.011889       0.075394      -0.2222   
average_listing_price_yy  180.0       0.071178       0.155735      -0.6842   
median_listing_price      180.0  297884.705556   89990.772264  158995.0000   
median_listing_price_mm   180.0       0.011084       0.085013      -0.3447   
median_listing_price_yy   180.0       0.087613       0.190472      -0.5571   
rural_%                   180.0      34.108520      19.131745       2.3638   
rural_cat                 180.0       1.800000       0.401116       1.0000   
total_listing_count       180.0    2579.777778    5400.795775       4.0000   

                                    25%            50%            75%  \
FIPS                        4007.000000    4013.000000    4021.000000   
year                        2019.000000    2019.000000    2019.000000   
density                        7.000000      16.000000      38.000000   
average_listing_price     266837.250000  395742.500000  473937.750000   
average_listing_price_mm      -0.011250       0.008200       0.021225   
average_listing_price_yy       0.003925       0.059150       0.118350   
median_listing_price      227267.750000  293884.500000  351125.000000   
median_listing_price_mm       -0.005725       0.005300       0.018800   
median_listing_price_yy        0.029825       0.068150       0.123650   
rural_%                       21.904623      33.197178      46.568686   
rural_cat                      2.000000       2.000000       2.000000   
total_listing_count          249.000000     742.000000    2075.000000   

                                    max  
FIPS                        4027.000000  
year                        2019.000000  
density                      480.000000  
average_listing_price     716258.000000  
average_listing_price_mm       0.727400  
average_listing_price_yy       0.624000  
median_listing_price      509000.000000  
median_listing_price_mm        0.916500  
median_listing_price_yy        1.820000  
rural_%                       74.061076  
rural_cat                      2.000000  
total_listing_count        26147.000000

In [19]:
# Checks the number of county
AZ_2019 = df_2019[(df_2019['state']=='AZ')]
AZ_2019['county'].drop_duplicates()

0          Apache County
12        Cochise County
24       Coconino County
36           Gila County
48         Graham County
60       Greenlee County
72         La Paz County
84       Maricopa County
96         Mohave County
108        Navajo County
120          Pima County
132         Pinal County
144    Santa Cruz County
156       Yavapai County
168          Yuma County
Name: county, dtype: object

In [20]:
# This is the number of counties in Hawaii
len(AZ_2019['county'].drop_duplicates())

15

In [21]:
AZ_2019_ALP = AZ_2019.groupby(['FIPS','county', 'density', 'rural_%', 
                               'rural_cat'])['average_listing_price'].mean().nlargest(15).reset_index()
AZ_2019_ALP

FIPS             county  density    rural_%  rural_cat  \
0   4013    Maricopa County      480   2.363800          2   
1   4005    Coconino County        7  31.466066          2   
2   4025     Yavapai County       29  33.197178          2   
3   4007        Gila County       11  41.062000          2   
4   4001      Apache County        5  74.061076          1   
5   4023  Santa Cruz County       38  26.883172          2   
6   4019        Pima County      113   7.523491          2   
7   4017      Navajo County       10  54.138242          1   
8   4015      Mohave County       16  22.963644          2   
9   4021       Pinal County       79  21.904623          2   
10  4012      La Paz County        3  56.327786          1   
11  4027        Yuma County       36  10.429576          2   
12  4003     Cochise County       20  36.301067          2   
13  4009      Graham County        8  46.437399          2   
14  4011    Greenlee County        5  46.568686          2   

    average_listing_price  
0           667522.250000  
1           666891.166667  
2           609945.666667  
3           471559.333333  
4           463216.250000  
5           444905.083333  
6           421055.583333  
7           391470.833333  
8           374457.583333  
9           325836.833333  
10          315948.583333  
11          254077.416667  
12          252562.333333  
13          241557.583333  
14          239195.500000

### 3.1b For the Year 2021

In [22]:
df_2021[(df_2021['state']=='AZ')].describe().T

count           mean            std         min  \
FIPS                      180.0    4013.866667       7.826613   4001.0000   
year                      180.0    2021.000000       0.000000   2021.0000   
density                   180.0      57.333333     117.154341      3.0000   
average_listing_price     180.0  594182.622222  208734.499933  68550.0000   
average_listing_price_mm  180.0       0.083623       0.824603     -0.7585   
average_listing_price_yy  180.0       0.405253       1.112464     -0.7952   
median_listing_price      180.0  382269.988889  129536.409398  75000.0000   
median_listing_price_mm   180.0       0.023181       0.095376     -0.3600   
median_listing_price_yy   180.0       0.205824       0.251501     -0.6999   
rural_%                   180.0      34.108520      19.131745      2.3638   
rural_cat                 180.0       1.800000       0.401116      1.0000   
total_listing_count       180.0    1652.472222    3307.001765      6.0000   

                                    25%            50%            75%  \
FIPS                        4007.000000    4013.000000    4021.000000   
year                        2021.000000    2021.000000    2021.000000   
density                        7.000000      16.000000      38.000000   
average_listing_price     433155.000000  587985.000000  784546.750000   
average_listing_price_mm      -0.019975       0.008800       0.039625   
average_listing_price_yy       0.099575       0.193350       0.333650   
median_listing_price      279975.000000  377747.500000  468150.000000   
median_listing_price_mm       -0.008525       0.015450       0.044650   
median_listing_price_yy        0.100725       0.183450       0.288075   
rural_%                       21.904623      33.197178      46.568686   
rural_cat                      2.000000       2.000000       2.000000   
total_listing_count          104.750000     430.500000    1538.500000   

                                   max  
FIPS                      4.027000e+03  
year                      2.021000e+03  
density                   4.800000e+02  
average_listing_price     1.151407e+06  
average_listing_price_mm  1.065310e+01  
average_listing_price_yy  1.092040e+01  
median_listing_price      7.025000e+05  
median_listing_price_mm   7.241000e-01  
median_listing_price_yy   1.728400e+00  
rural_%                   7.406108e+01  
rural_cat                 2.000000e+00  
total_listing_count       1.430400e+04

In [23]:
AZ_2021 = df_2021[(df_2021['state']=='AZ')]
AZ_2021_ALP = AZ_2021.groupby(['FIPS','county', 'density', 'rural_%', 
                               'rural_cat'])['average_listing_price'].mean().nlargest().reset_index()
AZ_2021_ALP

FIPS           county  density    rural_%  rural_cat  average_listing_price
0  4005  Coconino County        7  31.466066          2          899840.000000
1  4025   Yavapai County       29  33.197178          2          881942.916667
2  4013  Maricopa County      480   2.363800          2          833891.083333
3  4007      Gila County       11  41.062000          2          695058.083333
4  4011  Greenlee County        5  46.568686          2          687275.666667

In 2019, Arizona's Maricopa County has the highest average listing price. It is an urban with 2.26 rural percentage. The next highest average listing price were followed by Coconino, Yavapai, Gila, and Apache. In 2021, Maricopa County became third in the highest average lisiting price as Coconino County and Yavapai County demand became higher. Coconino, Yavapai, and Gila are urban, their rural percentages are higher than Maricopa. The former 5 placer, Apache, a rural area was replaced by Greenlee, an urban area with a higher rural percentage.<br>


## 3.2 Change Average Listing Price


### 3.2a For the Year 2019

In [24]:
AZ_2019_ALP_pct = AZ_2019.groupby(['FIPS','county', 'density', 'rural_%', 
                                   'rural_cat'])['average_listing_price_yy'].mean().nlargest().reset_index()
AZ_2019_ALP_pct

FIPS           county  density    rural_%  rural_cat  \
0  4001    Apache County        5  74.061076          1   
1  4009    Graham County        8  46.437399          2   
2  4012    La Paz County        3  56.327786          1   
3  4013  Maricopa County      480   2.363800          2   
4  4019      Pima County      113   7.523491          2   

   average_listing_price_yy  
0                  0.274367  
1                  0.182333  
2                  0.113650  
3                  0.094108  
4                  0.083492

### 3.2b For the Year 2021

In [25]:
AZ_2021_ALP_pct = AZ_2021.groupby(['FIPS','county', 'density', 'rural_%', 
                                   'rural_cat'])['average_listing_price_yy'].mean().nlargest().reset_index()
AZ_2021_ALP_pct

FIPS           county  density    rural_%  rural_cat  \
0  4011  Greenlee County        5  46.568686          2   
1  4012    La Paz County        3  56.327786          1   
2  4003   Cochise County       20  36.301067          2   
3  4017    Navajo County       10  54.138242          1   
4  4007      Gila County       11  41.062000          2   

   average_listing_price_yy  
0                  3.247392  
1                  0.374142  
2                  0.354325  
3                  0.337933  
4                  0.290417

In 2019, Apache, a rural area enjoyed the highest average listing price change. It is followed by Graham County, an urban with 45 rural %. Next is La Paz, a rural area which as a 56.33 rural % (not very high rural % compared to Apache's 74). In 2021, Greenlee, an urban area with 46.56 rural %) has the highest average listing price change 3.24% that tremendous increase compared to other counties of below .37% increase.<br>

In Arizona, there is an increase of demand for houses in the higher rural percentages in urban area such as Coconino and Yavapai. Both counties surpassed Maricopa in its higher average listing price in 2021. The top 5 average listing price change, Greenlee county, an urban area with 45.57 rural %, saw a tremendous percentage change, 325% compared with others that are below 40%.  


# 4. Individual States (Massachusetts)


## 4.1 Average Listing Price

### 4.1a For the Year 2019

In [26]:
# This are the list of counties in Massachusetts
MA_2019 = df_2019[(df_2019['state']=='MA')]
MA_2019['county'].drop_duplicates()

11505    Barnstable County
11517     Berkshire County
11529       Bristol County
11541         Essex County
11553      Franklin County
11565       Hampden County
11571     Hampshire County
11583     Middlesex County
11595       Norfolk County
11607      Plymouth County
11619       Suffolk County
11631     Worcester County
Name: county, dtype: object

In [27]:
# This is the number of counties in Massochusetts
len(MA_2019['county'].drop_duplicates())

12

In [28]:
MA_2019_ALP =MA_2019.groupby(['FIPS','county', 'density', 
                              'rural_%', 'rural_cat'])['average_listing_price'].mean().nlargest(12).reset_index()
MA_2019_ALP

FIPS             county  density    rural_%  rural_cat  \
0   25025     Suffolk County    13722   0.074091          2   
1   25017   Middlesex County     1995   3.027706          2   
2   25001  Barnstable County      581   7.483047          2   
3   25021     Norfolk County     1832   1.240489          2   
4   25009       Essex County     1644   4.244045          2   
5   25023    Plymouth County      805  10.307747          2   
6   25003   Berkshire County      139  31.593748          2   
7   25005     Bristol County     1047   9.834302          2   
8   25027   Worcester County      570  18.390286          2   
9   25015   Hampshire County      307  27.443699          2   
10  25011    Franklin County      101  54.431710          1   
11  25013     Hampden County      754   8.570843          2   

    average_listing_price  
0            1.471328e+06  
1            9.772631e+05  
2            9.664294e+05  
3            9.615623e+05  
4            8.081962e+05  
5            6.549784e+05  
6            6.314927e+05  
7            4.752997e+05  
8            4.286452e+05  
9            4.216835e+05  
10           3.365358e+05  
11           3.085722e+05

### 4.1b For the Year 2021

In [29]:
MA_2021 = df_2021[(df_2021['state']=='MA')]
MA_2021_ALP =MA_2021.groupby(['FIPS','county', 'density', 'rural_%', 
                              'rural_cat'])['average_listing_price'].mean().nlargest(12).reset_index()
MA_2021_ALP

FIPS             county  density    rural_%  rural_cat  \
0   25025     Suffolk County    13722   0.074091          2   
1   25001  Barnstable County      581   7.483047          2   
2   25021     Norfolk County     1832   1.240489          2   
3   25017   Middlesex County     1995   3.027706          2   
4   25003   Berkshire County      139  31.593748          2   
5   25009       Essex County     1644   4.244045          2   
6   25023    Plymouth County      805  10.307747          2   
7   25005     Bristol County     1047   9.834302          2   
8   25015   Hampshire County      307  27.443699          2   
9   25027   Worcester County      570  18.390286          2   
10  25011    Franklin County      101  54.431710          1   
11  25013     Hampden County      754   8.570843          2   

    average_listing_price  
0            1.582755e+06  
1            1.476872e+06  
2            1.083184e+06  
3            1.078144e+06  
4            9.752928e+05  
5            9.377348e+05  
6            7.506079e+05  
7            6.631889e+05  
8            5.355079e+05  
9            4.960157e+05  
10           4.360453e+05  
11           3.760823e+05

## 4.2 Change Average Listing Price


### 4.2 For the Year 2019

In [30]:
MA_2019_ALP_pct = MA_2019.groupby(['FIPS','county', 'density', 
                               'rural_%', 'rural_cat'])['average_listing_price_yy'].mean().nlargest(5).reset_index()
MA_2019_ALP_pct

FIPS            county  density    rural_%  rural_cat  \
0  25011   Franklin County      101  54.431710          1   
1  25003  Berkshire County      139  31.593748          2   
2  25025    Suffolk County    13722   0.074091          2   
3  25027  Worcester County      570  18.390286          2   
4  25009      Essex County     1644   4.244045          2   

   average_listing_price_yy  
0                  0.082025  
1                  0.080275  
2                  0.080050  
3                  0.061983  
4                  0.058617

### 4.2b For the Year 2021

In [31]:
MA_2021_ALP_pct =MA_2021.groupby(['FIPS','county', 'density', 'rural_%', 
                                  'rural_cat'])['average_listing_price_yy'].mean().nlargest(5).reset_index()
MA_2021_ALP_pct

FIPS             county  density    rural_%  rural_cat  \
0  25001  Barnstable County      581   7.483047          2   
1  25005     Bristol County     1047   9.834302          2   
2  25003   Berkshire County      139  31.593748          2   
3  25011    Franklin County      101  54.431710          1   
4  25015   Hampshire County      307  27.443699          2   

   average_listing_price_yy  
0                  0.277017  
1                  0.232325  
2                  0.211175  
3                  0.182767  
4                  0.137875

Pre-Covid and Post-Covid, Suffolk county maintained the highest average listing price. However, Middlesex county was displaced by Barnstable and Norfolk. Both of which have lower density and higher rural percentage than Middlesex. Berkshire county, an urban area with a high 31.59% rural percentage moved up to 5th place from 7th place. It surpassed Essex and Plythouth counties which have a higher density and lower rural percentage than Berkshire.<br>
Pre-COVID, the average listing price yearly change are below 10%. Post-COVID, the average listing price yearly change is in the 20%. It is headed by Barnstable county followed by Bristol county. Barnstable and Bristol counties are urban area with rural percentages below 10. Berkshire county has the 3rd highest average listing price yearly change. Berkshire county is an urban area with a high 31.59% rural percentage.
